eth_block.go

```diff
@@ -0,0 +1,363 @@
+	}
+
+	return ethTxNodes, ethTxTrieNodes
+}
+
+/*
+  OUTPUT
+*/
+
+// DecodeEthBlock takes a cid and its raw binary data
+// from IPFS and returns an EthBlock object for further processing.
+func DecodeEthBlock(c *cid.Cid, b []byte) (*EthBlock, error) {
+	var h types.Header
+	err := rlp.Decode(bytes.NewReader(b), &h)
+	if err != nil {
+		return nil, err
+	}
+
+	return &EthBlock{
+		Header:  &h,
+		cid:     c,
+		rawdata: b,
+	}, nil
+}
+
+/*
+  Block INTERFACE
+*/
+
+// RawData returns the binary of the RLP encode of the block header.
+func (b *EthBlock) RawData() []byte {
+	return b.rawdata
+}
+
+// Cid returns the cid of the block header.
+func (b *EthBlock) Cid() *cid.Cid {
+	return b.cid
+}
+
+// String is a helper for output
+func (b *EthBlock) String() string {
+	return fmt.Sprintf("<EthBlock %s>", b.cid)
+}
+
+// Loggable returns a map the type of IPLD Link.
+func (b *EthBlock) Loggable() map[string]interface{} {
+	return map[string]interface{}{
+		"type": "eth-block",
+	}
+}
+
+/*
+  Node INTERFACE
+*/
+
+// Resolve resolves a path through this node, stopping at any link boundary
+// and returning the object found as well as the remaining path to traverse
+func (b *EthBlock) Resolve(p []string) (interface{}, []string, error) {
+	if len(p) == 0 {
+		return b, nil, nil
+	}
+
+	switch p[0] {
+	case "bloom":
+		return b.Bloom, nil, nil // should not send the rest: You reached an end here.
+	case "coinbase":
+		return b.Coinbase, nil, nil
+	case "difficulty":
+		return b.Difficulty, nil, nil
+	case "extra":
+		return fmt.Sprintf("0x%x", b.Extra), nil, nil // This is a []byte. By default they are marshalled into Base64.
+	case "gaslimit":
+		return b.GasLimit, nil, nil
+	case "gasused":
+		return b.GasUsed, nil, nil
+	case "mixdigest":
+		return b.MixDigest, nil, nil
+	case "nonce":
+		return b.Nonce, nil, nil
+	case "number":
+		return b.Number, nil, nil
+	case "parent":
+		return &node.Link{Cid: commonHashToCid(MEthBlock, b.ParentHash)}, p[1:], nil
+	case "parentHash":
+		return b.ParentHash.Hex(), nil, nil
+	case "receipts":
+		return &node.Link{Cid: commonHashToCid(MEthTxReceiptTrie, b.ReceiptHash)}, p[1:], nil
+	case "receiptHash":
+		return b.ReceiptHash.Hex(), nil, nil
+	case "root":
+		return &node.Link{Cid: commonHashToCid(MEthStateTrie, b.Root)}, p[1:], nil
+	case "rootHash":
+		return b.Root.Hex(), nil, nil
+	case "time":
+		return b.Time, nil, nil
+	case "tx":
+		return &node.Link{Cid: commonHashToCid(MEthTxTrie, b.TxHash)}, p[1:], nil
+	case "txHash":
+		return b.TxHash.Hex(), nil, nil
+	case "uncles":
+		return &node.Link{Cid: commonHashToCid(MEthBlockList, b.UncleHash)}, p[1:], nil
+	case "uncleHash":
+		return b.UncleHash.Hex(), nil, nil
+	default:
+		return nil, nil, fmt.Errorf("no such link")
+	}
+}
+
+// Tree lists all paths within the object under 'path', and up to the given depth.
+// To list the entire object (similar to `find .`) pass "" and -1
+func (b *EthBlock) Tree(p string, depth int) []string {
+	return []string{
```

Comment:
Fixed in https://github.com/ipfs/go-ipld-eth/pull/5/commits/7cac47655bcc6028191a93302e2a9f33dd109b17